### PR TITLE
Eda, preprocessing and Sklearn models

### PR DESCRIPTION
**Pull request by JessicaOjo**
exploratory data analysis with visualizations on the graduate admission dataset

### CODE REVIEW

**Comment by JessicaOjo** [comment]
exploratory and pre-processing

**Comment by gogzicole** [comment]
1. this for me is the best EDA i've seen so far, the way you eased from numerical analysis of the datasets to their respective plots, was very eye catching.
2. Although; i don't necessarily fancy the choices of plots you used, the other EDA's had appropriate plots that are best suited for the illustration of the relationship of the dataset.
3. regardless your plots had the required labels/titles and brief summary of findings underneath each plot, which is desirable.
4. Lastly if you would be so nice as to delete the empty cell blocks at the bottom of your notebook, and also remove the comments on correlation, thank you.

**Comment by JessicaOjo** [comment]
Thank you Chigozie. I would work on it

**Comment by mickynnamdi** [comment]
I agree with Dozie comment

**Comment by gauravjain2** [comment]
Pretty good analysis 
Good Points:
- Well Labelled Plots
- Structured Code
- Inference from the plots also stated clearly

Critical Observations:
- After using the df.describe(), in the next code cells you are finding mean, max and min values, which are already present in the output of describe() function. So I think its merely a repetition of work 
- Do check for outliers in the EDA
- You can also work on Skewness of Data, Scaling and Normalization part also.

**Comment by JessicaOjo** [comment]
> Pretty good analysis
> Good Points:
> 
> * Well Labelled Plots
> * Structured Code
> * Inference from the plots also stated clearly
> 
> Critical Observations:
> 
> * After using the df.describe(), in the next code cells you are finding mean, max and min values, which are already present in the output of describe() function. So I think its merely a repetition of work
> * Do check for outliers in the EDA
> * You can also work on Skewness of Data, Scaling and Normalization part also.

Alright I’ll work on those and upload a new one

**Comment by JessicaOjo** [comment]
> Pretty good analysis
> Good Points:
> 
> * Well Labelled Plots
> * Structured Code
> * Inference from the plots also stated clearly
> 
> Critical Observations:
> 
> * After using the df.describe(), in the next code cells you are finding mean, max and min values, which are already present in the output of describe() function. So I think its merely a repetition of work
> * Do check for outliers in the EDA
> * You can also work on Skewness of Data, Scaling and Normalization part also.

Alright I’ll work on those and upload a new one

**Comment by blennyblex** [comment]
Nice one at Jessica, i went through your code line by line. The only issue i have with the model part is why are you performing metrics on the last 100 dataset(the vals, according to your codes), i thought those were meant for predictions only, and your metrics should be done on the training sets and not the last 100 meant for testing. if I'm wrong, I'm open to corrections

**Comment by JessicaOjo** [comment]
> Nice one at Jessica, i went through your code line by line. The only issue i have with the model part is why are you performing metrics on the last 100 dataset(the vals, according to your codes), i thought those were meant for predictions only, and your metrics should be done on the training sets and not the last 100 meant for testing. if I'm wrong, I'm open to corrections

Hello blenny, the instruction said to provide the errors on the validation set which should contain the last 100 entries, to achieve this I had to perform metrics on the validation set after each prediction.
It also helped me in monitoring the models for overfitting.
